src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -2084,12 +2103,95 @@ public EntryResult tMedicationDispense2MedicationDispense(
+			IBundleInfo localBundleInfo) {
+		EntryResult result = new EntryResult();
+
+		if (cdaSupplyOrder == null || cdaSupplyOrder.isSetNullFlavor())
+			return result;
+
+		MedicationRequest medRequest = new MedicationRequest();
+		result.addResource(medRequest);
+
+		// patient
+		medRequest.setSubject(getPatientRef());
+
+		// resource id
+		IdType resourceId = new IdType("MedicationRequest", getUniqueId());
+		medRequest.setId(resourceId);
+
+		// meta.profile
+		if (Config.isGenerateDafProfileMetadata())
+			medRequest.getMeta().addProfile(Constants.PROFILE_DAF_MEDICATION_REQUEST);
+
+		// id -> identifier
+		if (cdaSupplyOrder.getIds() != null && !cdaSupplyOrder.getIds().isEmpty()) {
+			for (II ii : cdaSupplyOrder.getIds()) {
+				if (ii != null && !ii.isSetNullFlavor()) {
+					medRequest.addIdentifier(dtt.tII2Identifier(ii));
+				}
+			}
+		}
+
+		// effectiveTime.low -> authoredOn
+		if (cdaSupplyOrder.getEffectiveTimes() != null && !cdaSupplyOrder.getEffectiveTimes().isEmpty()) {
+			for (SXCM_TS ts : cdaSupplyOrder.getEffectiveTimes()) {
+				if (ts instanceof IVL_TS) {
+					if (((IVL_TS) ts).getLow() != null && !((IVL_TS) ts).getLow().isSetNullFlavor()) {
+						medRequest.setAuthoredOnElement(dtt.tTS2DateTime(((IVL_TS) ts).getLow()));
+					}
+				}
+			}
+		}
+
+		// statusCode -> status
+		if (cdaSupplyOrder.getStatusCode() != null && !cdaSupplyOrder.getStatusCode().isSetNullFlavor()) {
+			medRequest.setStatus(vst.tActStatus2MedicationRequestStatus(cdaSupplyOrder.getStatusCode().getCode()));
+		}
+
+		// hardcoded to "instance-order"
+		medRequest.setIntent(MedicationRequestIntent.INSTANCEORDER);
+
+		// product.manufacturedProduct(MedicationInformation) -> medication
+		if (cdaSupplyOrder.getProduct() != null && !cdaSupplyOrder.getProduct().isSetNullFlavor()) {
+			EntryResult fhirMedicationResult = tManufacturedProduct2Medication(cdaSupplyOrder.getProduct());
+			if (fhirMedicationResult.hasResult()) {
+				// Should only have one result entry
+				BundleEntryComponent resultEntry = fhirMedicationResult.getBundle().getEntryFirstRep();
+				Medication medicationResult = (Medication) resultEntry.getResource();
+				result.addResource(medicationResult);
+				// We can only add either a reference here or a codeableconcept. Opting for
+				// Reference.
+				medRequest.setMedication(new Reference(medicationResult.getId()));
+			}
+		}
+
+		if (cdaSupplyOrder.getQuantity() != null) {
+			MedicationRequestDispenseRequestComponent dispenseRequest = new MedicationRequestDispenseRequestComponent();
+			medRequest.setDispenseRequest(dispenseRequest);
+
+			if (cdaSupplyOrder.getQuantity().getValue() != null) {
+				SimpleQuantity sq = new SimpleQuantity();
+				sq.setValue(cdaSupplyOrder.getQuantity().getValue());
+				sq.setUnit(cdaSupplyOrder.getQuantity().getUnit());
+				dispenseRequest.setQuantity(sq);
+			}
+
+			if (cdaSupplyOrder.getRepeatNumber() != null) {
```

Comment:
you should add repeatNumber to the null check at the front, if either of them comes in we want the object populated. Also, can you comment with the mappings here like it is in the other sections.

Reply:
Gotchya will do.